lib/cudnn/src/convolution.jl

```diff
@@ -264,15 +264,20 @@ end
 
 
 # Return algorithm with best memory that is within 10% of best time
-function cudnnConvolutionAlgoPerfChoose(convDesc, perfResults, n)
+function cudnnConvolutionAlgoPerfChoose(convDesc, tensorDesc, perfResults, n)
     mathType = Ref{cudnnMathType_t}(CUDNN_DEFAULT_MATH)
     cudnnGetConvolutionMathType(convDesc, mathType)
+    skipMathTypeCheck = begin  # See https://github.com/JuliaGPU/CUDA.jl/pull/1943#issuecomment-1605267932
+        dtype, _ = cudnnGetTensorDescriptor(tensorDesc)
```

Comment:
Per https://buildkite.com/julialang/cuda-dot-jl/builds/4082#0189195a-1428-4cf5-84b7-55376287aee3/275-443 this needs to handle different types of descriptors. Other than that the changes look fine to me.